src/rust/engine/process_executor/src/main.rs

```diff
@@ -462,7 +458,6 @@ async fn make_request_from_flat_args(
     concurrency_available: args.command.concurrency_available.unwrap_or(0),
     cache_scope: ProcessCacheScope::Always,
     execution_strategy: ProcessExecutionStrategy::Local,
-    platform_properties: collection_from_keyvalues(args.command.extra_platform_property.iter()),
```

Comment:
So, I think that these two arguments end up meaning that this will never execute a process remotely? It looks like this was previously converted to hardcode `ProcessExecutionStrategy::Local`, but it should probably be switching based on whether [a `--server` has been provided](https://github.com/pantsbuild/pants/blob/aa40c120284b09e60ffd10bc5f596e7f4655e99c/src/rust/engine/process_executor/src/main.rs#L148-L151)? The `process_executor` is really useful for testing remote execution.